### PR TITLE
Properly make these bindings a "header-only" (.di) library

### PR DESCRIPTION
**Pull request by CyberShadow**
- Fix remaining causes for an object file to be required
- Rename all `.d` files to `.di`

Follow-up to https://github.com/D-Programming-Deimos/openssl/pull/87

### CODE REVIEW

File: dub.sdl [hunk 1]
```diff
@@ -21,12 +21,3 @@ configuration "library-manual-version" {
...
 }
-
-configuration "unittest" {
-	targetType "executable"
-	targetName "openssl-test-library"
-	dflags "-main"
-	excludedSourceFiles "source/deimos/openssl/applink.d"
-	preGenerateCommands `${DUB} scripts/generate_version.d` platform="posix"
-	versions `DeimosOpenSSLAutoDetect`
-}
```
Comment (Geod24) [inline dub.sdl:32]: Have you checked what happened for a project depending on `openssl` ? Specifically, if it auto-detects the version or uses the fallback ?

**Comment by Geod24** [comment]
On the other hand, I greatly dislike the template approach for macros because it means they aren't type-checked (unless one decide to write unittests for all of them).

File: dub.sdl (hunk 1 above)
Comment (CyberShadow) [reply dub.sdl:32]: `dub test` in `examples/sslecho` seems to work just as fine as on master, does that count?

**Comment by CyberShadow** [comment]
Right, same as C macros, so we're feature-par here :)

I think we could write a test that enumerates and instantiates all such templates...

**Comment by kinke** [comment]
> Would this work with your downstream use case?

I've tested building the main project (executable) with this, and it compiles & links fine with both DMD and LDC on Linux at least. :+1:

File: dub.sdl (hunk 1 above)
Comment (Geod24) [reply dub.sdl:32]: I guess it does :)

**Comment by Geod24** [comment]
> Right, same as C macros, so we're feature-par here :)

Being feature-par with C is a pretty low bar... But let's go ahead with this.

**Pull request merged**